### PR TITLE
Maint/hazelcast properties

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Document Hazelcast for gateway and refactor properties.

**Reference**

Issues #179

### CODE REVIEW

File: refarch-gateway/README.md [hunk 1]
```diff
@@ -78,3 +80,28 @@ spring:
...
+
+### Hazelcast
+
+Beside the already mentioned properties Hazelcast also has the following requirements.
+
+#### Modular java
+See https://docs.hazelcast.com/hazelcast/5.5/getting-started/install-hazelcast#using-modular-java
+
+Following Java options need to be set.
+For the gateway image this can be done with `JAVA_OPTS_APPEND`.
```
Comment (hupling) [inline refarch-gateway/README.md:92]: im Helm Template heißt die Variable HAZELCAST_JAVA_OPTS_APPEND

File: refarch-gateway/README.md [hunk 2]
```diff
@@ -41,16 +41,18 @@ Beside the default behaviour there are some special route prefixes which are han
...
-| `SPRING_CLOUD_GATEWAY_ROUTES_<index>_FILTERS_<index>`    | List of filters applied to the route.              | `RewritePath=/api/backend-service/(?<urlsegments>.*), /$\{urlsegments}` |
-| `ALLOWED_ORIGINS_PUBLIC` (optional)                      | List of urls allowed as origin for public routes.  | `https://*.example.com,http://localhost:*`                              |
-| `ALLOWED_ORIGINS_CLIENTS` (optional)                     | List of urls allowed as origin for clients routes. | `https://*.example.com,http://localhost:*`                              |
-| `REFARCH_SECURITY_CSRFWHITELISTED_<index>` (optional)    | List of routes to disable csrf protection for.     | `/example/**`                                                           |
+| Var                                                      | Description                                                       | Example                                                                 |
+|----------------------------------------------------------|-------------------------------------------------------------------|-------------------------------------------------------------------------|
```
Comment (hupling) [inline refarch-gateway/README.md:45]: vielleicht ist es auch übersichtlicher irgendwie Unterüberschriften einziehen, sodass man die Konfigs gruppiert nach Funktion

**Review by devtobi — DISMISSED** [review]
LGTM

File: refarch-gateway/README.md (hunk 2 above)
Comment (simonhir) [reply refarch-gateway/README.md:45]: Think that's a good idea in general but didn't come up with any good splitting.
Think a heading for a single property is more unreadable as helping.

**Review by hupling — APPROVED** [review]
LGTM

File: refarch-gateway/src/main/java/de/muenchen/refarch/gateway/ApiGatewayApplication.java [hunk 3]
```diff
@@ -30,7 +31,7 @@
  * "https://cloud.spring.io/spring-cloud-gateway/reference/html/">https://cloud.spring.io/spring-cloud-gateway/reference/html/</a>
  */
 @SpringBootApplication
-@EnableConfigurationProperties(SecurityProperties.class)
+@EnableConfigurationProperties({ SecurityProperties.class, HazelcastProperties.class })
```
Comment (devtobi) [inline refarch-gateway/src/main/java/de/muenchen/refarch/gateway/ApiGatewayApplication.java:34]: Why are all Properties classes listed here separately. Shouldn't it be possible to make use of Spring automatically scanning the base package? e.g. via `ConfigurationPropertiesScan`?

File: refarch-gateway/README.md [hunk 4]
```diff
@@ -78,3 +79,28 @@ spring:
...
+For the gateway image this can be done with `JAVA_OPTS_APPEND`.
+```
+--add-modules java.se \
+  --add-exports java.base/jdk.internal.ref=ALL-UNNAMED \
+  --add-opens java.base/java.lang=ALL-UNNAMED \
+  --add-opens java.base/java.nio=ALL-UNNAMED \
+  --add-opens java.base/sun.nio.ch=ALL-UNNAMED \
+  --add-opens java.management/sun.management=ALL-UNNAMED \
+  --add-opens jdk.management/com.ibm.lang.management.internal=ALL-UNNAMED \
+  --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED
```
Comment (devtobi) [inline refarch-gateway/README.md:100]: Is setting this relevant for the `hazelcast-local` profile as well? Besides, I don't really understand the benefits of the `hazelcast-local` profile. Why should I need Hazelcast locally when I only run instance of the gateway?

File: refarch-gateway/src/main/java/de/muenchen/refarch/gateway/ApiGatewayApplication.java (hunk 3 above)
Comment (simonhir) [reply refarch-gateway/src/main/java/de/muenchen/refarch/gateway/ApiGatewayApplication.java]: Done

File: refarch-gateway/README.md (hunk 4 above)
Comment (simonhir) [reply refarch-gateway/README.md:100]: Theoretically yes but as always only one instance running not really required in my opinion.

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**